# The Battle of Neighborhoods
Capstone Project

## Hotels near CN Tower in Toronto

### **Introduction**

Let say there is a group of tourists would like go to Toronto and they would like to stay at good rating hotel near CN Tower since there are lots of good restaurants nearby. This analysis is used to find the best place to stay for them during their visit to Toronto.

### Data

The FourSquare API is used to collect data about locations of the various hotel nearby CN Tower, which later on be used for acquiring the information of the hotels and their ratting. 

### Methodology  
1. First, using the FourSquare API to explore the hotels near CN Tower  
2. Get the relevant information and transform it into a pandas dataframe  
3. Filter and summarize the hotels' information near CN Tower  
4. Visualize the hotels’ locations.  
5. Get the ratting for each hotel  
6. Summarize the results and find the best place to stay for the tourists

### Import Library needed

In [7]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation
!conda install -c conda-forge geopy --yes
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude
# libraries for displaying images
from IPython.display import Image
from IPython.core.display import HTML
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize
!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library
print('Folium installed')
print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.4.5.2 |       hecda079_0         147 KB  conda-forge
    geopy-1.22.0               |     pyh9f0ad1d_0          63 KB  conda-forge
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    certifi-2020.4.5.2         |   py36h9f0ad1d_0         152 KB  conda-forge
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.5 MB

The following NEW packages will be INSTALLED:

    geographiclib:   1.50-py_0           conda-forge
    geopy:          

### Explore hotel nearby CN Tower

In [8]:
CLIENT_ID = '1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO'
CLIENT_SECRET = 'ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO
CLIENT_SECRET:ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH


In [92]:
address = 'CN Tower, Toronto'
geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

43.6425637 -79.38708718320467


In [114]:
search_query = 'Hotel'
radius = 300 
print(search_query + ' .... OK!')

Hotel .... OK!


In [115]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO&client_secret=ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH&ll=43.6425637,-79.38708718320467&v=20180604&query=Hotel&radius=300&limit=30'

In [116]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5ee75a635c6a3e18e5f93b4c'},
 'response': {'venues': [{'id': '53c50853498e9a2b6b9464b5',
    'name': 'Marriott Toronto City Centre Hotel Fitness Center',
    'location': {'lat': 43.642092,
     'lng': -79.391362,
     'labeledLatLngs': [{'label': 'display',
       'lat': 43.642092,
       'lng': -79.391362}],
     'distance': 348,
     'cc': 'CA',
     'neighborhood': 'Entertainment District',
     'country': 'Canada',
     'formattedAddress': ['Canada']},
    'categories': [{'id': '4bf58dd8d48988d175941735',
      'name': 'Gym / Fitness Center',
      'pluralName': 'Gyms or Fitness Centers',
      'shortName': 'Gym / Fitness',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/gym_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1592220048',
    'hasPerk': False},
   {'id': '53357710498e20817350cfb4',
    'name': 'Delta Hotels by Marriott Toronto',
    'location': {'address': '75 Lower Simcoe Street',

In [117]:
# assign relevant part of JSON to venues
venues = results['response']['venues']
# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

categories  hasPerk  \
0  [{'id': '4bf58dd8d48988d175941735', 'name': 'G...    False   
1  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'H...    False   
2  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'H...    False   
3  [{'id': '4d954b06a243a5684965b473', 'name': 'R...    False   
4  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'H...    False   

                         id        location.address location.cc location.city  \
0  53c50853498e9a2b6b9464b5                     NaN          CA           NaN   
1  53357710498e20817350cfb4  75 Lower Simcoe Street          CA       Toronto   
2  5a495535ee71201850d17ad0         1 Blue Jays Way          CA       Toronto   
3  5261f967498e9055976bb354                   Tecom          CA         Dubai   
4  4b861600f964a5208e7f31e3   225 Front Street West          CA       Toronto   

  location.country location.crossStreet  location.distance  \
0           Canada                  NaN                348   
1           Canada                  NaN                255   
2           Canada                  NaN                175   
3           Canada                  NaN                225   
4           Canada                  NaN                247   

                           location.formattedAddress  \
0                                           [Canada]   
1  [75 Lower Simcoe Street, Toronto ON M5J 3A6, C...   
2      [1 Blue Jays Way, Toronto ON M5V 1J4, Canada]   
3                             [Tecom, Dubai, Canada]   
4  [225 Front Street West, Toronto ON M5V 2X3, Ca...   

                             location.labeledLatLngs  location.lat  \
0  [{'label': 'display', 'lat': 43.642092, 'lng':...     43.642092   
1  [{'label': 'display', 'lat': 43.6428819, 'lng'...     43.642882   
2  [{'label': 'display', 'lat': 43.643592, 'lng':...     43.643592   
3  [{'label': 'display', 'lat': 43.64142237945941...     43.641422   
4  [{'label': 'display', 'lat': 43.6444547, 'lng'...     43.644455   

   location.lng   location.neighborhood location.postalCode location.state  \
0    -79.391362  Entertainment District                 NaN            NaN   
1    -79.383949                     NaN             M5J 3A6             ON   
2    -79.388746                     NaN             M5V 1J4             ON   
3    -79.389398                     NaN                 NaN            NaN   
4    -79.385483                     NaN             M5V 2X3             ON   

                                                name    referralId  \
0  Marriott Toronto City Centre Hotel Fitness Center  v-1592220048   
1                   Delta Hotels by Marriott Toronto  v-1592220048   
2                 Toronto Marriott City Centre Hotel  v-1592220048   
3                                   Ramee Rose Hotel  v-1592220048   
4                    InterContinental Toronto Centre  v-1592220048   

  venuePage.id  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4    135876574

In [118]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name  \
0  Marriott Toronto City Centre Hotel Fitness Center   
1                   Delta Hotels by Marriott Toronto   
2                 Toronto Marriott City Centre Hotel   
3                                   Ramee Rose Hotel   
4                    InterContinental Toronto Centre   
5                                Soco Ballroom Foyer   
6                                   The Ritz-Carlton   
7                              FRHI Hotels & Resorts   

                                 categories  \
0                      Gym / Fitness Center   
1                                     Hotel   
2                                     Hotel   
3  Residential Building (Apartment / Condo)   
4                                     Hotel   
5                                  Ballroom   
6                                     Hotel   
7                           Coworking Space   

                                          address  cc     city country  \
0                                             NaN  CA      NaN  Canada   
1                          75 Lower Simcoe Street  CA  Toronto  Canada   
2                                 1 Blue Jays Way  CA  Toronto  Canada   
3                                           Tecom  CA    Dubai  Canada   
4                           225 Front Street West  CA  Toronto  Canada   
5  3rd floor, Delta Hotel, 75 Lower Simcoe Street  CA  Toronto  Canada   
6                      181 Wellington Street West  CA  Toronto  Canada   
7                      155 Wellington Street West  CA  Toronto  Canada   

   crossStreet  distance                                   formattedAddress  \
0          NaN       348                                           [Canada]   
1          NaN       255  [75 Lower Simcoe Street, Toronto ON M5J 3A6, C...   
2          NaN       175      [1 Blue Jays Way, Toronto ON M5V 1J4, Canada]   
3          NaN       225                             [Tecom, Dubai, Canada]   
4          NaN       247  [225 Front Street West, Toronto ON M5V 2X3, Ca...   
5          NaN       258  [3rd floor, Delta Hotel, 75 Lower Simcoe Stree...   
6  John Street       307  [181 Wellington Street West (John Street), Tor...   
7          NaN       374  [155 Wellington Street West, Toronto ON M5V 0C...   

                                      labeledLatLngs        lat        lng  \
0  [{'label': 'display', 'lat': 43.642092, 'lng':...  43.642092 -79.391362   
1  [{'label': 'display', 'lat': 43.6428819, 'lng'...  43.642882 -79.383949   
2  [{'label': 'display', 'lat': 43.643592, 'lng':...  43.643592 -79.388746   
3  [{'label': 'display', 'lat': 43.64142237945941...  43.641422 -79.389398   
4  [{'label': 'display', 'lat': 43.6444547, 'lng'...  43.644455 -79.385483   
5  [{'label': 'display', 'lat': 43.642727, 'lng':...  43.642727 -79.383880   
6  [{'label': 'display', 'lat': 43.64533015759588...  43.645330 -79.387089   
7  [{'label': 'display', 'lat': 43.64567660611423...  43.645677 -79.385340   

             neighborhood postalCode state                        id  
0  Entertainment District        NaN   NaN  53c50853498e9a2b6b9464b5  
1                     NaN    M5J 3A6    ON  53357710498e20817350cfb4  
2                     NaN    M5V 1J4    ON  5a495535ee71201850d17ad0  
3                     NaN        NaN   NaN  5261f967498e9055976bb354  
4                     NaN    M5V 2X3    ON  4b861600f964a5208e7f31e3  
5  Entertainment District    M5J 3A6    ON  5ad7f8a5c9f90779768db54a  
6                     NaN    M5V 3G7    ON  4d2ce8d1fd2e236a8ac8ebde  
7                     NaN    M5V 0C3    ON  541c283f498eb661ff11f1c0

In [120]:
#Filter only include chinese restaurant
df2= dataframe_filtered.loc[(dataframe_filtered.categories == "Hotel")]
df2

name categories                     address  \
1    Delta Hotels by Marriott Toronto      Hotel      75 Lower Simcoe Street   
2  Toronto Marriott City Centre Hotel      Hotel             1 Blue Jays Way   
4     InterContinental Toronto Centre      Hotel       225 Front Street West   
6                    The Ritz-Carlton      Hotel  181 Wellington Street West   

   cc     city country  crossStreet  distance  \
1  CA  Toronto  Canada          NaN       255   
2  CA  Toronto  Canada          NaN       175   
4  CA  Toronto  Canada          NaN       247   
6  CA  Toronto  Canada  John Street       307   

                                    formattedAddress  \
1  [75 Lower Simcoe Street, Toronto ON M5J 3A6, C...   
2      [1 Blue Jays Way, Toronto ON M5V 1J4, Canada]   
4  [225 Front Street West, Toronto ON M5V 2X3, Ca...   
6  [181 Wellington Street West (John Street), Tor...   

                                      labeledLatLngs        lat        lng  \
1  [{'label': 'display', 'lat': 43.6428819, 'lng'...  43.642882 -79.383949   
2  [{'label': 'display', 'lat': 43.643592, 'lng':...  43.643592 -79.388746   
4  [{'label': 'display', 'lat': 43.6444547, 'lng'...  43.644455 -79.385483   
6  [{'label': 'display', 'lat': 43.64533015759588...  43.645330 -79.387089   

  neighborhood postalCode state                        id  
1          NaN    M5J 3A6    ON  53357710498e20817350cfb4  
2          NaN    M5V 1J4    ON  5a495535ee71201850d17ad0  
4          NaN    M5V 2X3    ON  4b861600f964a5208e7f31e3  
6          NaN    M5V 3G7    ON  4d2ce8d1fd2e236a8ac8ebde

In [121]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=15) 


# add chinese restaurants as blue circle markers
for lat, lng, label in zip(df2.lat, df2.lng, df2.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=7,
        popup=label,
        fill=True,
        color='blue',
        fill_color='blue',
        fill_opacity=0.6
        ).add_to(venues_map)

# display map
venues_map

#### There are 4 hotels near CN Tower,which are: Delta Hotels, Toronto Mariott, Intercontinental and Ritz Carlton. Now, we want to find the ratting for each of them

#### Find ratting for Delta Hotels by Marriott Toronto	

In [124]:
venue_id = '53357710498e20817350cfb4' # ID of Delta Hotels 
url1 = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url1

'https://api.foursquare.com/v2/venues/53357710498e20817350cfb4?client_id=1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO&client_secret=ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH&v=20180604'

In [125]:
result = requests.get(url1).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'description', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '53357710498e20817350cfb4',
 'name': 'Delta Hotels by Marriott Toronto',
 'contact': {'phone': '4168491200',
  'formattedPhone': '(416) 849-1200',
  'twitter': 'deltatoronto',
  'facebook': '474384675994679',
  'facebookUsername': 'DeltaHotelsbyMarriottToronto',
  'facebookName': 'Delta Hotels by Marriott Toronto'},
 'location': {'address': '75 Lower Simcoe Street',
  'lat': 43.6428819,
  'lng': -79.3839491,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.6428819,
    'lng': -79.3839491}],
  'postalCode': 'M5J 3A6',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['75 Lower Simcoe Street',
   'Toronto ON M5J 3A6',
   'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/delta-hotels-by-marriott-toronto/53357710498e20817350cfb4',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hote

In [126]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

9.0


#### Find ratting for Toronto Marriott City Centre Hotel	

In [137]:
venue_id = '5a495535ee71201850d17ad0' # ID of Toronto Mariott
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/5a495535ee71201850d17ad0?client_id=1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO&client_secret=ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH&v=20180604'

In [138]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']


dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'seasonalHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto'])


{'id': '5a495535ee71201850d17ad0',
 'name': 'Toronto Marriott City Centre Hotel',
 'contact': {},
 'location': {'address': '1 Blue Jays Way',
  'lat': 43.643592,
  'lng': -79.388746,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.643592,
    'lng': -79.388746}],
  'postalCode': 'M5V 1J4',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['1 Blue Jays Way', 'Toronto ON M5V 1J4', 'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/toronto-marriott-city-centre-hotel/5a495535ee71201850d17ad0',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_',
    'suffix': '.png'},
   'primary': True}],
 'verified': False,
 'stats': {'tipCount': 0},
 'likes': {'count': 7,
  'groups': [{'type': 'others',
    'count': 7,
    'items': [{'id': '220953413',
      'firstName': 'Josette',
      'lastName':

In [140]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.4


#### Find ratting for InterContinental Toronto Centre

In [143]:
venue_id = '4b861600f964a5208e7f31e3' # ID of intercontinental
url= 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4b861600f964a5208e7f31e3?client_id=1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO&client_secret=ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH&v=20180604'

In [144]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'hasMenu', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'venuePage', 'reasons', 'description', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4b861600f964a5208e7f31e3',
 'name': 'InterContinental Toronto Centre',
 'contact': {'phone': '4165971400',
  'formattedPhone': '(416) 597-1400',
  'twitter': 'ictorontocentre',
  'facebook': '209386829102400',
  'facebookUsername': 'intercontinentaltorontocentre',
  'facebookName': 'InterContinental Toronto Centre'},
 'location': {'address': '225 Front Street West',
  'lat': 43.6444547,
  'lng': -79.3854827,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.6444547,
    'lng': -79.3854827}],
  'postalCode': 'M5V 2X3',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['225 Front Street West',
   'Toronto ON M5V 2X3',
   'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/intercontinental-toronto-centre/4b861600f964a5208e7f31e3',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel

In [145]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

7.3


#### Find ratting for Ritz Carlton

In [146]:
venue_id = '4d2ce8d1fd2e236a8ac8ebde' # ID of Ritz Carlton
url= 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
url

'https://api.foursquare.com/v2/venues/4d2ce8d1fd2e236a8ac8ebde?client_id=1ISHINJHFE1IC2FK0WAWM31JNQS1TD2HOFWCEIOVU12RGFQO&client_secret=ZSX0TXEXOCJ15MZQR4FAZG1ZKJW20UHZDV2XKGEIWBNDAZZH&v=20180604'

In [147]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'seasonalHours', 'defaultHours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4d2ce8d1fd2e236a8ac8ebde',
 'name': 'The Ritz-Carlton',
 'contact': {'phone': '4165852500',
  'formattedPhone': '(416) 585-2500',
  'twitter': 'ritzcarlton',
  'facebook': '146358378747962',
  'facebookUsername': 'ritzcarltontoronto',
  'facebookName': 'The Ritz-Carlton, Toronto'},
 'location': {'address': '181 Wellington Street West',
  'crossStreet': 'John Street',
  'lat': 43.645330157595886,
  'lng': -79.38708901405334,
  'labeledLatLngs': [{'label': 'display',
    'lat': 43.645330157595886,
    'lng': -79.38708901405334}],
  'postalCode': 'M5V 3G7',
  'cc': 'CA',
  'city': 'Toronto',
  'state': 'ON',
  'country': 'Canada',
  'formattedAddress': ['181 Wellington Street West (John Street)',
   'Toronto ON M5V 3G7',
   'Canada']},
 'canonicalUrl': 'https://foursquare.com/v/the-ritzcarlton/4d2ce8d1fd2e236a8ac8ebde',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https://ss3.4sqi.

In [148]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

8.6


### RESULT

### from our calculation, we now get the ratting for each hotels: 
| Hotels       | Ratting  
| ------------- |:-------------:  
| Delta Hotels by Marriott Toronto      | 9.0
| Toronto Marriott City Centre Hotel|7.4
|InterContinental Toronto Centre	|7.3
| Ritz Carlton | 8.6

### Recommendation and Conclusion
From the results above, Delta hotels by Marriott Toronto has the highest ratting and it is located near CN Tower. Hence we recommend the tourists to stay at **Delta Hotels by Marriott Toronto**.  
However, this result is solely based on the location and ratting, assuming the price is not a matter for the tourists, if the tourists want to include the price as a factor, then further study need to be conducted.
